server/engine/src/main/java/org/eclipse/lsp/cobol/implicitDialects/sql/MessageServiceParser.java

```diff
@@ -122,6 +124,57 @@ protected void validateIntegerRange(String input, Integer minValue, Integer maxV
     }
   }
 
+  /**
+   * Validate LOB size and throw an error if it is incorrect
+   *
+   * @param input integer to check
+   * @param maxValue allowed integer value
+   */
+  protected void validateLobSize(Token start, Token stop, String input, Integer maxValue) {
+    if (input != null) {
+      Pattern pattern =
+          Pattern.compile("(?i)(CLOB|DBCLOB|BLOB)\\s*\\(\\s*(\\d+)(?:\\s*([KMG]))?\\s*\\)");
```

Comment:
This is compiled for every single check and it does not match e.g. `CHARACTER LARGE OBJECT`